mllib/src/main/scala/org/apache/spark/mllib/classification/StreamingLogisticRegressionWithSGD.scala

```diff
@@ -101,4 +107,14 @@ class StreamingLogisticRegressionWithSGD private[mllib] (
     this.model = Some(algorithm.createModel(initialWeights, 0.0))
     this
   }
+
+  override def setDecayFactor(decayFactor: Double): this.type = {
```

Comment:
This boilerplate is duplicated in streaming linear regression. I am guessing you do this to get the concrete subclass (correct me if I'm wrong), but you actually don't need to do this since the `this.type` in `trait StreamingDecay` takes care of this. A simple REPL example:

```scala
scala> trait Superclass { def test: this.type }
defined trait Superclass

scala> class Subclass extends Superclass { def test = this }
defined class Subclass

scala> (new Subclass()).test
res0: Subclass = Subclass@1cb4ab3e
```